### PR TITLE
Pass framerate from input stream_format or options to output stream format

### PR DESCRIPTION
**Pull request by Noarkhh**
- Pass framerate from input stream_format or options to output stream format

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 Membrane H264 parser.
 It is the Membrane element responsible for parsing the incoming h264 stream. The parsing is done as a sequence of the following steps:
-* splitting the h264 stream into stream NAL units, based on the "Annex B" of the "ITU-T Rec. H.264 (01/2012)"
+* splitting the h264 stream into stream NAL units, based on the "ITU-T Rec. H.264 (01/2012)"
```
Comment (varsill) [inline README.md:9]:
```suggestion
* splitting the H264 stream into stream NAL units, based on the "ITU-T Rec. H.264 (01/2012)"
```
We can stick to that formatting ;)

Comment (varsill) [reply README.md:9]: I believe that the splitting might be also based on length-prefix, which is not defined in ITU-T Rec. H.264 (01/2012)

File: lib/membrane_h264_plugin/parser.ex [hunk 2]
```diff
@@ -181,16 +181,20 @@ defmodule Membrane.H264.Parser do
...
+      else
+        {nil, nil}
+      end
 
     state = %{
       nalu_splitter: nil,
       nalu_parser: nil,
       au_splitter: AUSplitter.new(),
       au_timestamp_generator: au_timestamp_generator,
+      framerate: framerate,
```
Comment (varsill) [inline lib/membrane_h264_plugin/parser.ex:197]: I am just slightly worried that we have the `framerate` that might change based on the `stream_format` received, and we use it to create an output `stream_format`, and the `generate_best_effort_timestamps.framerate` that we set once, via options,  and use to generate timestamps

File: lib/membrane_h264_plugin/parser.ex [hunk 3]
```diff
@@ -589,32 +595,17 @@ defmodule Membrane.H264.Parser do
...
-          end
-
-        metadata =
-          if i == 0 do
-            put_in(metadata, [:metadata, :h264, :new_access_unit], %{key_frame?: is_keyframe?})
-          else
-            metadata
-          end
-
-        {metadata, nalu_start + byte_size(nalu.payload)}
```
Comment (varsill) [inline lib/membrane_h264_plugin/parser.ex:615]: I am slightly surprised that `nalu_start` is not used anywhere :D

Comment (Noarkhh) [reply lib/membrane_h264_plugin/parser.ex:615]: It was used to generate `prefixed_poslen` and `unprefixed_poslen`, but i didn't remove it then

File: lib/membrane_h264_plugin/parser.ex (hunk 2 above)
Comment (Noarkhh) [reply lib/membrane_h264_plugin/parser.ex:197]: The only case is when the incoming `H264` stream_format has `framerate` set to nil and `generate_best_effort_timestamps` was provided

**Review by varsill — APPROVED** [review]

**Pull request merged**